### ISSUE TITLE
ChainBuildException - Cycle detected after #39352 PR

### ISSUE THREAD

**Issue by rsvoboda** [quarkusio/quarkus#39468]
### Describe the bug

I noticed regression with recent Quarkus main using https://github.com/quarkus-qe/beefy-scenarios 002-quarkus-all-extensions module

```
[ERROR] Errors:
[ERROR]   AlmostAllQuarkusExtensionsTest.testQuarkusEndpointWithManyExtensions » Runtime java.lang.RuntimeException: io.quarkus.builder.ChainBuildException: Cycle detected:
		   io.quarkus.arc.deployment.ArcProcessor#signalBeanContainerReady produced class io.quarkus.arc.deployment.BeanContainerBuildItem
		to io.quarkus.flyway.deployment.FlywayProcessor#startActions produced class io.quarkus.agroal.spi.JdbcDataSourceSchemaReadyBuildItem
		to io.quarkus.hibernate.orm.deployment.metrics.HibernateOrmMetricsProcessor#metrics produced class io.quarkus.deployment.metrics.MetricsFactoryConsumerBuildItem
		to io.quarkus.smallrye.metrics.deployment.SmallRyeMetricsProcessor#extensionMetrics produced class io.quarkus.arc.deployment.UnremovableBeanBuildItem
		to io.quarkus.arc.deployment.ArcProcessor#validate produced class io.quarkus.arc.deployment.ValidationPhaseBuildItem
		to io.quarkus.arc.deployment.ArcProcessor#generateResources produced class io.quarkus.arc.deployment.ResourcesGeneratedPhaseBuildItem
		to io.quarkus.arc.deployment.ArcProcessor#initializeContainer produced class io.quarkus.arc.deployment.ArcContainerBuildItem
		to io.quarkus.arc.deployment.ArcProcessor#notifyBeanContainerListeners produced class io.quarkus.arc.deployment.PreBeanContainerBuildItem
		to io.quarkus.arc.deployment.ArcProcessor#signalBeanContainerReady
```

I was able to pin down the PR that caused this regression and it's https://github.com/quarkusio/quarkus/pull/39352 CC @geoand 

I have prepared minimal reproducer application [code-with-quarkus.zip](https://github.com/quarkusio/quarkus/files/14614002/code-with-quarkus.zip) to identify the minimal set of dependencies :
```xml
    <dependency>
      <groupId>io.quarkus</groupId>
      <artifactId>quarkus-resteasy-reactive</artifactId>
    </dependency>
    <dependency>
      <groupId>io.quarkus</groupId>
      <artifactId>quarkus-hibernate-orm</artifactId>
    </dependency>
    <dependency>
      <groupId>io.quarkus</groupId>
      <artifactId>quarkus-flyway</artifactId>
    </dependency>
    <dependency>
      <groupId>io.quarkus</groupId>
      <artifactId>quarkus-smallrye-metrics</artifactId>
    </dependency>
```

The key part is quarkus-smallrye-metrics usage, mentioned PR fixes issue for Hibernate / Micrometer combination but introduces regression for SRye Metrics. 

### Expected behavior

Build ends succesfully

### Actual behavior

Build fails

### How to Reproduce?

Run `mvn clean verify` in 002-quarkus-all-extensions module from https://github.com/quarkus-qe/beefy-scenarios 

Or run `mvn clean verify` with this minimal reproducer application [code-with-quarkus.zip](https://github.com/quarkusio/quarkus/files/14614002/code-with-quarkus.zip)

### Output of `uname -a` or `ver`

macOS

### Output of `java -version`

Java 21

### Quarkus version or git rev

Quarkus main d30f1618e3f1c6c3d1effc0180bb58db8cad49f0

**Comment by geoand** [comment]
Lovely...

@jmartisk @ebullient given that this problem does not occur with Micrometer but only with Smallrye Metrics, is there any the latter could be fixed to not cause this problem?

**Comment by jmartisk** [comment]
I've submitted a PR that seems to fix it: https://github.com/quarkusio/quarkus/pull/39470

**Comment by geoand** [comment]
Thanks!

**Comment by polarctos** [comment]
I think we also see this error with `3.8.3` now, where `3.8.2` still worked:
````
Failed to start quarkus: java.lang.RuntimeException: io.quarkus.builder.ChainBuildException: Cycle detected:
                   io.quarkus.arc.deployment.ArcProcessor#validate produced class io.quarkus.arc.deployment.ValidationPhaseBuildItem
                to io.quarkus.hibernate.orm.panache.deployment.PanacheHibernateResourceProcessor#validate produced class io.quarkus.arc.deployment.ValidationPhaseBuildItem$ValidationErrorBuildItem
                to io.quarkus.arc.deployment.ArcProcessor#generateResources produced class io.quarkus.arc.deployment.ResourcesGeneratedPhaseBuildItem
                to io.quarkus.arc.deployment.ArcProcessor#initializeContainer produced class io.quarkus.arc.deployment.ArcContainerBuildItem
                to io.quarkus.arc.deployment.ArcProcessor#notifyBeanContainerListeners produced class io.quarkus.arc.deployment.PreBeanContainerBuildItem
                to io.quarkus.arc.deployment.ArcProcessor#signalBeanContainerReady produced class io.quarkus.arc.deployment.BeanContainerBuildItem
                to io.quarkus.flyway.deployment.FlywayProcessor#startActions produced class io.quarkus.agroal.spi.JdbcDataSourceSchemaReadyBuildItem
                to io.quarkus.hibernate.orm.deployment.metrics.HibernateOrmMetricsProcessor#metrics produced class io.quarkus.deployment.metrics.MetricsFactoryConsumerBuildItem
                to io.quarkus.smallrye.metrics.deployment.SmallRyeMetricsProcessor#extensionMetrics produced class io.quarkus.arc.deployment.UnremovableBeanBuildItem
                to io.quarkus.arc.deployment.ArcProcessor#validate
````

@gsmet As the applied milestones annotated here are 3.9.0 and 3.10, will there be a patch release for this for 3.8 LTS too?

**Comment by geoand** [comment]
The fix will be applied to the next 3.8 release
